### ISSUE TITLE
multitenant: SHOW RANGES leaks adjacent tenant spans

### ISSUE THREAD

**Issue by ecwall** [cockroachdb/cockroach#94377]
In a cluster with multiple tenants, the output of `SHOW RANGES` for one tenant will display other tenants' spans
```
        tc := testCase{
		system: tenantExpected{
			result: [][]string{
				{"<before:/Table/104/1/1>", "…/1"},
				{"…/1", "<after:/Tenant/10>"},
			},
		},
		secondaryNoCapability: tenantExpected{
			result: [][]string{
				{"<before:/Tenant/10/Table/104/1/1>", "…/104/2/1"},
				{"…/104/2/1", "<after:/Tenant/11>"},
			},
		},
		secondaryCapability: tenantExpected{
			result: [][]string{
				{"<before:/Tenant/11/Table/104/1/1>", "…/104/2/1"},
				{"…/104/2/1", "<after:/Max"},
			},
		},
	}
```
https://github.com/cockroachdb/cockroach/pull/94314/files#diff-f7f3f42a43ce63bb6a1479e91b24673827957dab6f52d4a018d50c47e1e56a35R547

**Comment by knz** [comment]
Slack thread: https://cockroachlabs.slack.com/archives/C0KB9Q03D/p1675795955408409

TLDR: we are not going to change the low-level split points, and the logical exclusive end key for tenant 10 will continue to be the first key for tenant 11. However, we will want to update the *display* of these keys in `crdb_internal.ranges{,no_lease}` and the output of SHOW RANGES, to translate `/Tenant/<NextID>` to `/Tenant/<curID/Max` during pretty-printing.
